### PR TITLE
core: honor FileDescriptorName= too for Accept=yes sockets, plus several other cleanups

### PR DESCRIPTION
**Pull request by YHNdnzj**

### CODE REVIEW

File: src/core/socket.c [hunk 1]
```diff
@@ -422,6 +422,9 @@ static int socket_verify(Socket *s) {
         if (!strv_isempty(s->symlinks) && !socket_find_symlink_target(s))
                 return log_unit_error_errno(UNIT(s), SYNTHETIC_ERRNO(ENOEXEC), "Unit has symlinks set but none or more than one node in the file system. Refusing.");
 
+        if (s->accept && s->fdname)
+                log_unit_warning(UNIT(s), "FileDescriptorName= has no effect in combination with Accept=yes. Ignoring.");
```
Comment (yuwata) [inline src/core/socket.c]: Instead, why not to set the specified name? I don't think it breaks existing system, unless mistakenly set the name in .socket but the triggered service explicitly expects name with "connection".

Comment (poettering) [reply src/core/socket.c]: yeah, i was wondering the same. i.e. what was the rationale for not propagating that name in the first place? i mean, i guess it was me who implemented it like that, but i cannot come up with a reason why. i am inclined to just make this work, it's much simpler that way.

or can you guess what i was thinking back then?

File: src/core/unit.c [hunk 2]
```diff
@@ -4216,7 +4216,7 @@ static int user_from_unit_name(Unit *u, char **ret) {
         return 0;
 }
 
-static int unit_verify_contexts(const Unit *u, const ExecContext *ec) {
+static int unit_verify_contexts(const Unit *u, const ExecContext *ec, const KillContext *kc) {
```
Comment (poettering) [inline src/core/unit.c]: i'd probably do the reverse here: drop both ec and kc from this function prototype and instead try to get both contexts locally via unit_get_exec_context() and unit_get_kill_context().

I think that would be simpler.

**Comment by poettering** [comment]
So, unless somehone figures out why in heaven i didn't inherit down the configured fdname when for accepted connections, I think we should just make that work. It's a minor compat break. but one of the kind we probably should be able to get away with, as long as we document it in NEWS.

(please add this to NEWS then)

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**